great_expectations/experimental/datasources/interfaces.py

```diff
@@ -77,7 +80,7 @@ class BatchSorter:
     reverse: bool = False
 
 
-BatchSortersDefinition: TypeAlias = Union[List[BatchSorter], List[str]]
+BatchSortersDefinition: TypeAlias = Union[List[BatchSorter], List[str], list]
```

Comment:
This typing info is now passed to the API-level. We need to add the empty `list` since we use `order_by = sorters or []`.

Reply:
I would have expected assigning an empty list to a `List[str]` would be fine.
```python
my_list: List[str] = [] # this should be ok
```

Reply:
It has something to do with the `Union` in the `TypeAlias`:

```
my_list_alias: TypeAlias = List[str]
my_list: Optional[my_list_alias] = None
my_list_arg = my_list or []
reveal_type(my_list_arg)  # Revealed type is "builtins.list[builtins.str]"
```

```
my_list_alias: TypeAlias = Union[List[str], List[int]]
my_list: Optional[my_list_alias] = None
my_list_arg = my_list or []
reveal_type(my_list_arg)  # Revealed type is "Union[builtins.list[builtins.str], builtins.list[builtins.int], builtins.list[Any]]"
```

Possibly a `mypy` bug...

Reply:
Decision was made to stop using the `TypeAlias` and use `Union[List[str], List[int]]` where the alias is currently being used.

Reply:
@Kilo59 I figured it out. Instead of:
`TypeAlias = Union[List[BatchSorter], List[str]]` it should be `TypeAlias = List[Union[BatchSorter, str]]`